src/context/service-worker-context.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, useEffect, useState } from 'react'
+import { registerServiceWorker } from '../service-worker-utils'
+
+export const ServiceWorkerContext = createContext({
+  isServiceWorkerRegistered: false
+})
+
+export const ServiceWorkerProvider = ({ children }): JSX.Element => {
+  const [isServiceWorkerRegistered, setIsServiceWorkerRegistered] = useState(false)
+
+  useEffect(() => {
+    if (isServiceWorkerRegistered) {
+      return
+    }
+    async function doWork (): Promise<void> {
+      const registration = await navigator.serviceWorker.getRegistration()
+
+      if (registration != null) {
+        // service worker already registered
+        // attempt to update it
+        await registration.update()
+        setIsServiceWorkerRegistered(true)
+      } else {
+        try {
+          await registerServiceWorker()
+          setIsServiceWorkerRegistered(true)
+        } catch (err) {
+          // eslint-disable-next-line no-console
+          console.error('config-debug: error registering service worker', err)
+        }
+      }
+    }
+    void doWork()
```

Comment:
I'm curious, is this just to suppress floating promises?

Reply:
since there are a number of promises being worked on in there, I used this method, rather than .then().catch() promise nesting & chaining